src/Tvheadend.cpp

```diff
@@ -1419,18 +1415,12 @@ void CTvheadend::ParseRecordingAddOrUpdate ( htsmsg_t *msg, bool bAdd )
   /* Error */
   if ((str = htsmsg_get_str(msg, "error")) != NULL)
   {
-    if (!strcmp(str, "300"))
-    {
-      UPDATE(rec.state, PVR_TIMER_STATE_ABORTED);
-    }
-    else if (strstr(str, "missing") != NULL)
+    if (rec.IsTimer())
```

Comment:
Sorry, but this will not work. "File missing" error string is set by tvh for a finished recording (https://github.com/tvheadend/tvheadend/blob/master/src/htsp_server.c#L720). 

Setting PVR_TIMER_STATE_ERROR for the recording with this particular error (having state PVR_TIMER_STATE_COMPLETED in this particular situation) is important as it makes Kodi put this recording in the Timer list in the GUI, which is fine.

With your change, it will keep the bogus recording in PVR_TIMER_STATE_COMPLETED which makes it appear in the list of finished recordings, which is wrong per definition.